engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java

```diff
@@ -370,50 +370,10 @@ boolean equalsInteropObjectWithMembers(
       @Shared("typesLib") @CachedLibrary(limit = "10") TypesLibrary typesLib,
       @Shared("equalsNode") @Cached EqualsNode equalsNode,
       @Shared("hostValueToEnsoNode") @Cached HostValueToEnsoNode valueToEnsoNode) {
-    try {
-      Object selfMembers = interop.getMembers(selfObject);
```

Comment:
@Akirathan, do you remember why you wrote this code? I can't find a test relying on it except 386f3bf43f51baa47fb87d8eb491371386aa690c which seems erroneous and against the spec - functions (a constructor with not fully saturated arguments) shall never be equal.

Reply:
About the test - it is possible there is no test for this case. I guess that I should have created a foreign JS or Python object with some members to test this case. 

In 386f3bf43f51baa47fb87d8eb491371386aa690c - I am not sure I understand why that should be erroneous. `Date.new hours=3` is not a function, but a `Date` atom, no?

Reply:
No, it is not atom. The method doesn't have `hours` parameter - e.g. it is a `Function` with over-saturated value `hours=3` and missing value `year=_`.

Reply:
Oh, you are right. I mismatched it with `Date_Time`. But still, `Date.new year=1973` *is* a `Date` atom. Therefore, `Date.new year=1973 == Date.new year=1973` does not test `equalsInteropObjectWithMembers`, but `EqualsAtomNode`

Reply:
Now, the new version of the test uses `EqualsIsAtomNode`. The previous version of the test was falling back to `getMembers()` of `Function` as it is also a `TruffleObject` and has members (`EQUALS`), it has been resolved as equal!

Now that problem is fixed: [38d0553](https://github.com/enso-org/enso/pull/7780/commits/38d0553476ffd6196911e0a1bef2f01b146eb567)